annotation-processor/src/main/java/com/hubspot/annotation/processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,151 @@
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import com.google.auto.service.AutoService;
+
+@SupportedAnnotationTypes("java.lang.SuppressWarnings")
+@AutoService(Processor.class)
+public class ModernizerAnnotationProcessor extends AbstractProcessor {
+
+    @Override
+    public final SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latest();
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        for (TypeElement annotation : annotations) {
+            List<String> annotatedClasses =
+                getAnnotatedClasses(roundEnv, annotation);
+            makeAnnotatedClassesFile(annotatedClasses);
+        }
+        return true;
+    }
+
+    private List<String> getAnnotatedClasses(
+        RoundEnvironment roundEnv, TypeElement annotation) {
+        List<String> annotatedClasses = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        FileWriter writer = null;
+        try {
+            File dir = new File(getFilePath(), "modernizer");
+            if (!dir.exists()) {
+                dir.mkdir();
+            }
+            File file = new File(dir.getPath(),
+                "modernizer-ignore-annotated-classes.txt");
+            writer = new FileWriter(file);
+            for (String element : annotatedClasses) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private String getFilePath() {
+        try {
+            FileObject fileObjectToGetPath =
+                processingEnv.getFiler().createResource(
+                    StandardLocation.CLASS_OUTPUT,
+                    "",
+                    "dummy-file.txt",
+                    null);
+            String path = fileObjectToGetPath.getName();
+            fileObjectToGetPath.delete();
+            while (!path.endsWith("target")) {
+                path = path.substring(0, path.lastIndexOf("/"));
```

Comment:
This code will likely change, but operating on a File using `getParentFile` would be best here. Similarly, the loop condition could use `getName().equals("target")` instead of `endswith.`